cpp/cmake/thirdparty/CUDF_GetLibcudacxx.cmake

```diff
@@ -0,0 +1,33 @@
+#=============================================================================
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_and_configure_libcudacxx VERSION)
+    CPMFindPackage(NAME     libcudacxx
+        VERSION             ${VERSION}
+        GIT_REPOSITORY      https://github.com/NVIDIA/libcudacxx.git
+        GIT_TAG             ${VERSION}
+        GIT_SHALLOW         TRUE
+        DOWNLOAD_ONLY       TRUE
+    )
+    set(LIBCUDACXX_DIR "${libcudacxx_SOURCE_DIR}" PARENT_SCOPE)
+    set(LIBCUDACXX_INCLUDE_DIR "${libcudacxx_SOURCE_DIR}/include" PARENT_SCOPE)
+    set(LIBCXX_DIR "${libcudacxx_SOURCE_DIR}/libcxx" PARENT_SCOPE)
+    set(LIBCXX_INCLUDE_DIR "${libcudacxx_SOURCE_DIR}/libcxx/include" PARENT_SCOPE)
```

Comment:
Are these because libcu++ doesn't expose a target? I think we should define our own libcu++ interface target and depend on that.

Reply:
`libcudacxx` is the target name, but I kept these as-is instead of changing everywhere we're currently using them.

Reply:
I'm not sure I follow. If there's a `libcudacxx` target, why do we need to do anything other than `target_link_libraries( cudf libcudacxx)`?

Reply:
Yeah that takes care of adding the include paths to the target, but we use these variables in our stringification module too: https://github.com/rapidsai/cudf/blob/db210ffc4b46a775a0ed9a1313b5f0d48bad3704/cpp/cmake/Modules/StringifyJITHeaders.cmake#L167-L168

We could hard-code paths like `${libcudacxx_SOURCE_DIR}/libcxx/include` into the places where the variables are used, but the variables seem marginally better than magic paths in more places.

Reply:
Ah, the stringification is what makes our use case unique. Got it.